controllers/operatorpipeline_controller.go

```diff
@@ -64,6 +65,10 @@ func (r *OperatorPipelineReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, err
 	}
 
+	// setting pipeline pointer on reconciler struct to allow usage
+	// on subreconciler functions
+	r.OperatorPipeline = pipeline
```

Comment:
This seems like a bad idea from a concurrency perspective. I'd have to do some investigation, but first glance seems like a problem.
Scenario: Two reconciles happen back to back. Before the first reconcile is complete, the second occurs and overwrites this field. The first reconcile is now in an inconsistent state, as it was operating on a different CR for half of it's reconcile. 

IIUC, there is only one Reconciler. Now, if you wanted to use a mutex, that would be fine.

Reply:
As we discussed today, there is way more behind the scenes than a simple struct loading the data. Workers are quite independent of each other. But since it's a little bit obscure I also agree to go through the most comfortable path here. BTW thanks for the help with the code and all tips on rebasing this.